pkg/proxy/userspace/proxier.go

```diff
@@ -465,12 +479,15 @@ func (proxier *Proxier) addServiceOnPortInternal(service proxy.ServicePortName,
 	proxier.serviceMap[service] = si
 
 	klog.V(2).Infof("Proxying for service %q on %s port %d", service, protocol, portNum)
-	go func(service proxy.ServicePortName, proxier *Proxier) {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
 		defer runtime.HandleCrash()
-		atomic.AddInt32(&proxier.numProxyLoops, 1)
+		wg.Done()
 		sock.ProxyLoop(service, si, proxier.loadBalancer)
-		atomic.AddInt32(&proxier.numProxyLoops, -1)
-	}(service, proxier)
+	}()
+	// Wait until the ProxyLoop goroutine has actually started
+	wg.Wait()
```

Comment:
I'm not sure why this is being added. Does this actually do anything other than slow down the return a little? This still can return before `sock.ProxyLoop` is called, which doesn't seem fundamentally different in nature from what it was doing before.

Reply:
@liggitt yeah, I guess that doesn't need to be there; all we really care about is that the socket is open and the iptables rules have been added by openPortal(), so I removed that bit and squashed the fixup.